### ISSUE TITLE
Don't expand large gists - leave as a link

### ISSUE THREAD

**Issue by kbrock** [gitterHQ/gitter#1399]
When I want to share code ~5 lines, I put it in a code block
When I want to share something bigger, I put it in a gist.

But gitter pastes the gist into the window.

I would like gitter to not paste the code into the window if it is bigger than some small threshold (possibly 5 lines)

It would also be nice to be able to collapse this code that was auto included

**Comment by MadLittleMods** [comment]
Here is a contrary issue https://github.com/gitterHQ/gitter/issues/1217

To get around this, you can link some inline code of the URL. Not ideal but a way to squash the embed for others.

```
[`https://gist.github.com/MadLittleMods/0e38f03774fb16e8d698175e505f1f3e`](https://gist.github.com/MadLittleMods/0e38f03774fb16e8d698175e505f1f3e)
```

Having a comprehensive preference/setting to not automatically expand embeds or even some sort of customizable threshold height to just never show would be great on a user preference level: https://github.com/gitterHQ/gitter/issues/714

**Comment by kbrock** [comment]
@MadLittleMods thanks for the work around (as always)

**Comment by darsain** [comment]
Came here to complain about the same thing since I can't take it anymore. Embedding every link to its 640px height limit is very annoying. Everyone I see discussing it hates it. People that encounter it for the first time always apologize for what they did (i.e. pollute the whole screen with their link), remove the link, try to paste the code to some service they think won't expand, just to be surprised when it does again... chaos all around.

As it was said above, if people want to post small snippets, they just paste them. If they want to past big ones, they use a dedicated service so they won't pollute the whole channel screen. Expanding these links goes precisely against the intention of the link.

I don't know who would want this as a default behavior. Don't know a single person that doesn't curse whenever gitter expands big blocks of code. From this perspective I just don't understand the motivation behind #1217.

### Suggestion

Disabling it with a checkbox would be a quick one, but better solution might be adding a [+] button right after the link that will expand its contents below the message. The expanded block should still be capped at 640px, but be scrollable when it overflows, so it's actually useful! The current version hides the rest of the embed, so you are forced to deal with the polluted screen AND have to click the link to see its contents anyway.

In other words: expand on demand.

This behavior could be also extend to code snippets. They should be capped at 200px tops with their bottom border being a thin toggle button, which will expand to the reasonable 640px limit and become scrollable when it overflows. As a bonus, when the block is collapsed, make the whole snippet an expand button.

With code blocks working like this, people might not even need 3rd party services to paste big chunks of code, leading to a more streamlined and comfy UX.

**Comment by kbrock** [comment]
Lol. Thanks @darsain

I also can not fathom that anyone would want this expanded.

Yet, here we are. Such a fun fun world.
